test/sshkit/sshkit_test.exs

```diff
@@ -6,80 +6,153 @@ defmodule SSHKitTest do
+      context = SSHKit.context(["10.0.0.1", "10.0.0.2"])
+
+      hosts = [
+        %Host{name: "10.0.0.1", options: []},
+        %Host{name: "10.0.0.2", options: []}
+      ]
+
+      assert context == %Context{hosts: hosts}
     end
 
-    test "creates with list of Maps" do
+    test "creates a context from a list of maps with :name and :options" do
+      context =
+        SSHKit.context([
+          %{name: "10.0.0.1", options: [user: "me"]},
+          %{name: "10.0.0.2", options: []}
+        ])
+
       hosts = [
-        %{name: "10.0.0.1", options: []},
-        %{name: "10.0.0.2", options: []},
-        %{name: "10.0.0.3", options: []}
+        %Host{name: "10.0.0.1", options: [user: "me"]},
+        %Host{name: "10.0.0.2", options: []}
       ]
-      context = SSHKit.context(hosts)
 
-      assert context == @context_simple
+      assert context == %Context{hosts: hosts}
     end
 
-    test "creates with mixed list" do
+    test "creates a context from a list of tuples" do
+      context =
+        SSHKit.context([
+          {"10.0.0.1", [user: "me"]},
+          {"10.0.0.2", []}
+        ])
+
       hosts = [
-        "10.0.0.1",
-        %{name: "10.0.0.2", options: []},
-        %Host{name: "10.0.0.3", options: []}
+        %Host{name: "10.0.0.1", options: [user: "me"]},
+        %Host{name: "10.0.0.2", options: []}
       ]
-      context = SSHKit.context(hosts)
 
-      assert context == @context_simple
+      assert context == %Context{hosts: hosts}
     end
 
-    test "includes shared options" do
-      hosts = ["10.0.0.1", "10.0.0.2", "10.0.0.3"]
-      context = SSHKit.context(hosts, @options)
-      assert context == @context_options
+    test "creates a context from a mixed list" do
+      context =
+        SSHKit.context([
+          %Host{name: "10.0.0.4", options: [user: "me"]},
+          %{name: "10.0.0.3", options: [password: "three"]},
+          {"10.0.0.2", port: 2222},
+          "10.0.0.1"
+        ])
+
+      hosts = [
+        %Host{name: "10.0.0.4", options: [user: "me"]},
+        %Host{name: "10.0.0.3", options: [password: "three"]},
+        %Host{name: "10.0.0.2", options: [port: 2222]},
+        %Host{name: "10.0.0.1", options: []}
+      ]
+
+      assert context == %Context{hosts: hosts}
     end
 
-    test "merges shared options" do
+    test "includes shared options" do
+      context =
+        [{"10.0.0.1", user: "me"}, "10.0.0.2"]
+        |> SSHKit.context(port: 2222)
+
       hosts = [
-        %{name: "10.0.0.1", options: [password: "123"]},
-        "10.0.0.2",
-        "10.0.0.3"
+        %Host{name: "10.0.0.1", options: [port: 2222, user: "me"]},
+        %Host{name: "10.0.0.2", options: [port: 2222]}
       ]
-      context = SSHKit.context(hosts, @options)
-      assert context == @context_merged_options
+
+      assert context == %Context{hosts: hosts}
     end
 
     test "does not override host options with shared options" do
```

Comment:
Maybe speak about "default options" here instead - now that the naming changed 🤔

Reply:
Makes perfect sense!

As a side note: I wasn't 100% sure about the rename, but – to me – "default" makes it clearer that individual host options override these when comparing to "shared options".

Reply:
Yep, fully agree 👍 

As said, I actually didn't fully understand the original implementation 🙈